include/swift/Basic/OwnedString.h

```diff
@@ -60,20 +65,25 @@ class OwnedString {
     else
       this->Data = Data;
   }
-
+  OwnedString(const char* Data, size_t Length, StringOwnership Ownership) {
+    initialize(Data, Length, Ownership);
+  }
 public:
-  OwnedString() : OwnedString(nullptr, 0, StringOwnership::Unowned) {}
+  OwnedString(): OwnedString(nullptr, 0, StringOwnership::Unowned) {}
 
-  OwnedString(const char *Data, size_t Length)
-    : OwnedString(Data, Length, StringOwnership::Unowned) {}
+  OwnedString(const char *Data, size_t Length):
+    OwnedString(Data, Length, StringOwnership::Unowned) {}
 
   OwnedString(StringRef Str) : OwnedString(Str.data(), Str.size()) {}
 
   OwnedString(const char *Data) : OwnedString(StringRef(Data)) {}
 
-  OwnedString(const OwnedString &Other)
-      : OwnedString(Other.Data, Other.Length, Other.Ownership) {}
-  
+  OwnedString(const OwnedString &Other) {
+    // Release previously allocated data.
+    release();
```

Comment:
This can't happen. Are you sure it didn't mean the copy assignment operator?

(Also, you should probably add move operators as well. [Rule of Five](http://en.cppreference.com/w/cpp/language/rule_of_three)!)

Reply:
Oh, you’re right. This meant to be the copy assignment operator. Will fix